### PR TITLE
Implement the Scones paper

### PR DESCRIPTION
**Pull request by jvamvas**
This PR implements Scones, which is described in the following paper: [Jam or Cream First? Modeling Ambiguity in Neural Machine Translation with SCONES](https://aclanthology.org/2022.naacl-main.365) (Stahlberg & Kumar, NAACL 2022).

At training time, provide the argument `--scones` to enable the objective, and optionally set `--scones-alpha` to balance the positive and negative loss terms. The resulting model is automatically configured to be `multi_label_model=True`
At inference time, a multi-label model will automatically use the sigmoid activation instead of the default softmax.

The initial author of the code changes is @fhieber. I have made some changes to make sure that the implementation covers all features of Sockeye.

#### Pull Request Checklist ##
- [x] Changes are complete (if posting work-in-progress code, prefix your pull request title with '[WIP]'
until you can check this box.
- [x] Unit tests pass (`pytest`)
- [x] Were system tests modified? If so did you run these at least 5 times to account for the variation across runs?
- [x] System tests pass (`pytest test/system`)
- [x] Passed code style checking (`./style-check.sh`)
- [x] You have considered writing a test
- [ ] Updated major/minor version in `sockeye/__init__.py`. Major version bump if this is a backwards incompatible change.
- [ ] Updated CHANGELOG.md

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sockeye/beam_search.py [hunk 1]
```diff
@@ -1071,7 +1077,8 @@ def get_search_algorithm(models: List[SockeyeModel],
                          prevent_unk: bool = False,
                          greedy: bool = False,
                          skip_nvs: bool = False,
-                         nvs_thresh: Optional[float] = None) -> Union[BeamSearch, GreedySearch]:
+                         nvs_thresh: Optional[float] = None,
+                         use_sigmoid: bool = False) -> Union[BeamSearch, GreedySearch]:
```
Comment (fhieber) [inline sockeye/beam_search.py:1081]: Looks like indentation is one off

**Review by fhieber — COMMENTED** [review]
Thanks! Looks good apart from one nitpick :)

**Review by mjdenkowski — COMMENTED** [review]
Thanks for implementing this!

How difficult is it to install JAX and run the sample code for SCONES from the paper? It would be great to include a unit test that compares the output of our loss functions against pre-computed values from the JAX implementation. The test could be modeled after our [cross-entropy loss test](https://github.com/awslabs/sockeye/blob/main/test/unit/test_loss.py#L68-L98).

File: sockeye/beam_search.py (hunk 1 above)
Comment (fhieber) [reply sockeye/beam_search.py:1081]: looks okay now. Resolving.

**Comment by mjdenkowski** [comment]
Closing for inactivity. Please feel free to update/reopen.
